definitions/procedures/foreman/fix_nic_type.rb

```diff
@@ -0,0 +1,35 @@
+module Procedures::Foreman
+  class FixNicType < ForemanMaintain::Procedure
+    metadata do
+      advanced_run true
+      description 'Fix NIC type from Nic::Base to Nic::Managed'
+    end
+
+    def run
+      nics = feature(:foreman_database).query(sql_select)
+
+      return unless nics.any?
+
+      answer = ask_decision("Found #{nics.size} interfaces with 'Nic::Base' type.\n\n" \
+       'Do you want update them to Nic::Managed type?', actions_msg: 'y(yes), q(quit)')
+
+      abort! unless answer == :yes
+
+      update_base_nics
+    end
+
+    private
+
+    def sql_select
+      <<-SQL
+        SELECT id FROM nics WHERE type = 'Nic::Base'
```

Comment:
Whilst reproducing this with @adamruzicka we've noticed that it's possible to create (in not supported way, but Rails/Foreman is okay with that) `Nic` with empty `type`, which can be considered in similar way as invalid. I guess we could catch here more false-valid types: `nil`, `""`, `Nic::Base`, `Nic::Interface`.

Reply:
Just got my hands on a reproducer machine, the offending records had `NULL` type